utilities/setup_dev_bucket.py

```diff
@@ -0,0 +1,351 @@
+def copy_folder(
+    source_bucket: str,
+    target_bucket: str,
+    path: str,
+) -> None:
+    """Copies folder from a production bucket to a dev bucket"""
+    assert target_bucket.startswith(BUCKET_PREFIX)
+    logger.info(f"Copying {source_bucket}/{path} to ./tmp")
+    s3.download_folder(
+        source_bucket, prefix=path, export_path=Path("./tmp/dev_bucket_setup")
+    )
+    logger.info(f"Copying ./tmp to {target_bucket}/{path}")
+    s3.upload_folder(
+        target_bucket,
+        Path(f"./tmp/dev_bucket_setup/{path}"),
+        Path(path),
+        "private",
+        contents_only=True,
+    )
+
+
+def resolve_latest_recipe(
+    target_bucket: str,
+    ds: str,
+):
+    """
+    Within a dev bucket, resolves/copies the dataset currently in "latest"
+    into its versioned folder under "datasets/{dataset_id}"
+    """
+    os.environ["RECIPES_BUCKET"] = target_bucket
+    importlib.reload(configuration)
+    assert configuration.RECIPES_BUCKET != PROD_RECIPES_BUCKET
+    input = recipes.Dataset(name=ds, version="latest")
+    resolved = recipes.Dataset(name=ds, version=recipes.get_latest_version(ds))
+    os.environ["RECIPES_BUCKET"] = PROD_RECIPES_BUCKET
+    importlib.reload(configuration)
+    assert target_bucket.startswith(BUCKET_PREFIX)
+    s3.copy_folder(
+        target_bucket,
+        input.s3_folder_key("datasets") + "/",
+        resolved.s3_folder_key("datasets") + "/",
+        "private",
+    )
+
+
+def resolve_latest_publish(target_bucket: str, data_product: str):
+    """
+    Within a dev bucket, resolves/copies the data product currently in "publish/latest"
+    into its versioned folder within "{product}/publish"
+    """
+    os.environ["PUBLISHING_BUCKET"] = target_bucket
+    importlib.reload(configuration)
+    assert configuration.PUBLISHING_BUCKET != PROD_PUBLISHING_BUCKET
+    input = PublishKey(product=data_product, version="latest")
+    latest = publishing.get_latest_version(data_product)
+    assert latest
+    resolved = PublishKey(product=data_product, version=latest)
+    os.environ["PUBLISHING_BUCKET"] = PROD_PUBLISHING_BUCKET
+    importlib.reload(configuration)
+    s3.copy_folder(
+        target_bucket,
+        input.path + "/",
+        resolved.path + "/",
+        "private",
+    )
+
+
+def clone_recipe(
+    target_bucket: str,
+    dataset_id: str,
+    version: str = "latest",
+):
+    """
+    Given a dataset id, copy it from edm-recipes
+    If provided version is "latest", also resolve it to its versioned folder
+    """
+    key = DatasetKey(name=dataset_id, version=version)
+    copy_folder(
+        source_bucket=configuration.RECIPES_BUCKET,
+        target_bucket=target_bucket,
+        path=f"{key.s3_path("datasets")}/",
+    )
+    if version == "latest":
+        resolve_latest_recipe(target_bucket, dataset_id)
+
+
+def clone_recipes_latest(
+    target_bucket: str,
+    datasets: list[str] | None = None,
+    dataset_mode: Literal["include", "exclude"] | None = None,
```

Comment:
I think a kwargs for `exclude_datasets` and `include_datasets` would be a little more straightforward.

Reply:
That is far more sensible